lib/puppet/functions/yaml_data.rb

```diff
@@ -30,10 +30,8 @@ def yaml_data(options, context)
           Puppet.warning(msg)
           {}
         end
-      rescue YAML::SyntaxError => ex
-        # Psych errors includes the absolute path to the file, so no need to add that
-        # to the message
-        raise Puppet::DataBinding::LookupError, "Unable to parse #{ex.message}"
+      rescue Puppet::Util::Yaml::YamlLoadError => ex
+        raise Puppet::DataBinding::LookupError, "Unable to parse (%{path}): %{message}" % { path: path, message: ex.message }
```

Comment:
Comparing this rescue to the one at https://github.com/puppetlabs/puppet/pull/6972/files#diff-52884bb8683e87a10f77bbbdf7575f71R61 (where the comment is retained). If the comment is true, then this change is invalid. If not, then it should be applied there too.

Reply:
The comment was correct before this PR. `YAML.{load,safe_load}` includes the path if there is a syntax error:

```
$ ruby -ryaml -e "YAML.load(File.read('bad.yaml'), 'bad.yaml')"
/usr/local/opt/rbenv/versions/2.4.3/lib/ruby/2.4.0/psych.rb:377:in `parse': (bad.yaml): did not find expected node content while parsing a flow node at line 2 column 1 (Psych::SyntaxError)
```

But `YAML.safe_load` does not when raising `Psych::DisallowedClass`:

```
$ ruby -ryaml -e "YAML.safe_load(File.read('illegal.yaml'), [], [], false, 'illegal.yaml')"
/usr/local/opt/rbenv/versions/2.4.3/lib/ruby/2.4.0/psych/class_loader.rb:97:in `find': Tried to load unspecified class: Object (Psych::DisallowedClass)
```

I think `Puppet::Util::Yaml` should handle the Psych exceptions differently and raise the appropriate message.